### DISCUSSION TITLE
Encoded transaction as argument.

### DISCUSSION THREAD

**Discussion by ghost** [CoolBitX-Technology/coolwallet-signing-tools#100] [Q&A] [answered]
We are integrating Algorand. Algorand transactions have 30 - 40 transaction fields and most of them are optional. The fields are Message Pack encoded before signing. Public key to Address will require SHA512/256. Since the script assembler doesnt have support for this MessagePack encoding and the required hashing methods and I guess 40 transction fields with optional nature will require a many combination of signing scripts. Is it alright if the encoded transaction along with the information to be displayed in card for confirmation are sent as argument to the card and get the signature for the transaction.

**Comment by hankliu418** [comment, marked as answer]
For security consideration, we will not provide display information independent with signing data. We are now working on implement Message Pack in se card, will release later.
